### PR TITLE
Always show sidebar toggle, also when sidebar is open

### PR DESCRIPTION
**Pull request by jancborchardt**
This makes the interface more predictable and doesn’t leave the fullscreen toggle hanging in mid-air in some settings.

@Ivansss @danxuliu we only need to add the handler to make the sidebar-trigger close the sidebar again. I guess it’s in sidebarview.js but don’t know fully how, can you help me here? :)

### CODE REVIEW

**Comment by danxuliu** [comment]
Clicking on the sidebar toggle when the sidebar is closed opens it. But clicking on it when it is open does nothing. That is a bit strange.

However, clicking on the sidebar toggle to close it... also feels a bit strange for me :-P Probably it is just because I am already used to it (also from the Files app), but the most natural way to close the sidebar is using the _Close_ button. However, if the sidebar toggle closes the sidebar, should the _Close_ button be removed? Or could it be kept?

In any case, if the sidebar toggle _toggles_ the sidebar then its appearance should reflect this and it should have a higher opacity or something like that when the sidebar is open.

**Comment by jancborchardt** [comment]
@danxuliu that’s why I wrote:

> we only need to add the handler to make the sidebar-trigger close the sidebar again. I guess it’s in sidebarview.js but don’t know fully how, can you help me here? :)

;)

> Probably it is just because I am already used to it (also from the Files app), but the most natural way to close the sidebar is using the Close button.

There’s no natural way. With the left navigation, we have the the menu toggle as well. Apart from clicking the menu toggle again, clicking on the empty part below where the content is works again, to go back to the content. Or of course when you click a navigation entry which then gets loaded in the content.

> However, if the sidebar toggle closes the sidebar, should the Close button be removed? Or could it be kept?

When we fix the width of the right sidebar as per https://github.com/nextcloud/spreed/pull/569 (instead of filling whole space on mobile) so the button is visible when the sidebar is open, we can do the same here as for the left navigation. And yes, probably remove the close button. Which I would like a lot since it interferes with content elements like the call name, name editing, and copy link functions.

**Comment by danxuliu** [comment]
@jancborchardt 
>  that’s why I wrote:
>
>  >    we only need to add the handler to make the sidebar-trigger close the sidebar again. I guess it’s in sidebarview.js but don’t know fully how, can you help me here? :)

:facepalm: Sorry, my bad ;-)

>  When we fix the width of the right sidebar as per #569

To be honest I hope that it will not be merged :-P But I do not understand, how is that related to this pull request? As long as the sidebar toggle is always shown next to the sidebar, why does it matter whether the size is fixed or not?

>  And yes, probably remove the close button. Which I would like a lot since it interferes with content elements like the call name, name editing, and copy link functions.

OK, no problem; I can live without _Close_ button :-P

**Comment by danxuliu** [comment]
Just for your information, I have been working on the missing part, but I found a related issue in server that is taking me longer than expected to fix :-(

**Comment by jancborchardt** [comment]
> To be honest I hope that it will not be merged :-P But I do not understand, how is that related to this pull request? As long as the sidebar toggle is always shown next to the sidebar, why does it matter whether the size is fixed or not?

The problem is that on mobile, the sidebar will take up 100% of width. Hence we can not remove the x Close button, since otherwise you can not exit the sidebar again because the sidebar toggle left _outside_ of the sidebar will not be shown.

**Comment by nickvergessen** [comment]
will take over

**Comment by nickvergessen** [comment]
Can't be handled the same way as the fullscreen icon, so reopening

File: js/views/participantview.js [hunk 1]
```diff
@@ -139,7 +139,7 @@
 					return '<span><div class="avatar" data-user="' + escapeHTML(element.id) + '" data-user-display-name="' + escapeHTML(element.displayName) + '"></div>' + escapeHTML(element.displayName) + '</span>';
 				},
 				formatSelection: function () {
-					return '<span class="select2-default" style="padding-left: 0;">'+OC.L10N.translate('spreed', 'Choose person…')+'</span>';
+					return '<span class="select2-default" style="padding-left: 0;">'+OC.L10N.translate('spreed', 'New call …')+'</span>';
```
Comment (nickvergessen) [inline js/views/participantview.js]: This is wrong, it adds a user to the participant list.
Anyway the naming is fixed in https://github.com/nextcloud/spreed/pull/751

Should not be done in a PR that deals with "sidebar toggle"

**Comment by Ivansss** [comment]
Rebased to master

**Comment by Ivansss** [comment]
Didn't remove the close button because when app is in 'fullscreen' the sidebar overlays chat view and ('fullscreen' & 'sidebar-toggle') buttons.
This should be fixed in another PR.

**Comment by nickvergessen** [comment]
> The position of the toggle button needs to be fixed in public pages due to nextcloud/server#9002 That will be done in a different pull request, though, as that should not be backported while the rest of this pull request should.

@danxuliu can you do this?

**Comment by danxuliu** [comment]
>  can you do this?

Yes, I will do that.

**Pull request merged**